public/scripts/reasoning.js

```diff
@@ -161,6 +167,48 @@ function registerReasoningSlashCommands() {
             return message.extra.reasoning;
         },
     }));
+
+    SlashCommandParser.addCommandObject(SlashCommand.fromProps({
+        name: 'reasoning-parse',
+        returns: 'reasoning string',
+        helpString: t`Extracts the reasoning block from a string using the Reasoning Formatting settings.`,
+        namedArgumentList: [
+            SlashCommandNamedArgument.fromProps({
+                name: 'regex',
+                description: 'Whether to apply regex scripts to the reasoning content.',
+                typeList: ARGUMENT_TYPE.BOOLEAN,
+                defaultValue: 'true',
+                isRequired: false,
```

Comment:
enum provider for boolean is sadly still not automatically applied. I would add it here as enumList, like the other places